research/gaia/gremlin/gremlin_core/src/structure/property.rs

```diff
@@ -23,15 +23,74 @@ use std::io;
 use std::ops::{Deref, DerefMut};
 use std::sync::Arc;
 
+pub type PropId = u32;
+
 #[derive(Clone, Eq, PartialEq, Hash, Debug)]
 pub enum Token {
     Id,
     Label,
-    Property(String),
+    Property(PropKey),
+}
+
+#[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub enum PropKey {
+    Str(String),
+    Id(PropId),
+}
+
+impl From<String> for PropKey {
+    fn from(prop_name: String) -> Self {
+        PropKey::Str(prop_name)
+    }
+}
+
+impl From<&String> for PropKey {
+    fn from(prop_name: &String) -> Self {
+        PropKey::Str(prop_name.clone())
+    }
+}
+
+impl From<i32> for PropKey {
```

Comment:
This should be `From<PropId>` instead of `From<i32>` ?

Reply:
Done